framework/src/main/java/org/checkerframework/framework/util/dependenttypes/DependentTypesHelper.java

```diff
@@ -149,304 +162,209 @@ public boolean hasDependentAnnotations() {
         }
         List<String> elements = new ArrayList<>();
         for (Method method : methods) {
-            org.checkerframework.framework.qual.JavaExpression javaExpression =
+            org.checkerframework.framework.qual.JavaExpression javaExpressionAnno =
                     method.getAnnotation(org.checkerframework.framework.qual.JavaExpression.class);
-            if (javaExpression != null) {
+            if (javaExpressionAnno != null) {
                 elements.add(method.getName());
             }
         }
         return elements;
     }
 
-    ///
-    /// Viewpoint adaptation
-    ///
+    /**
+     * Returns the elements of the annotation that are Java expressions.
+     *
+     * @param am AnnotationMirror
+     * @return the elements of the annotation that are Java expressions
+     */
+    private List<String> getListOfExpressionElements(AnnotationMirror am) {
+        for (Class<? extends Annotation> clazz : annoToElements.keySet()) {
+            if (factory.areSameByClass(am, clazz)) {
+                return annoToElements.get(clazz);
+            }
+        }
+        return Collections.emptyList();
+    }
+
+    /**
+     * Creates a TreeAnnotator that viewpoint-adapts dependent type annotations.
+     *
+     * @return a new TreeAnnotator that viewpoint-adapts dependent type annotations
+     */
+    public TreeAnnotator createDependentTypesTreeAnnotator() {
+        assert hasDependentAnnotations();
+        return new DependentTypesTreeAnnotator(factory, this);
+    }
 
     /**
      * Viewpoint-adapts the dependent type annotations on the bounds to the use of the type.
      *
      * @param classDecl class or interface declaration whose type variables should be viewpoint
      *     adapted
-     * @param bounds annotated types of the bounds of the type variables; side-effected by this
-     *     method
+     * @param bounds annotated types of the bounds of the type variables; its elements are
+     *     side-effected by this method (but the list itself is not side-effected)
      */
-    public void viewpointAdaptTypeVariableBounds(
+    public void atTypeVariableBounds(
```

Comment:
This is an odd name because it viewpoint-adapts to a use, not to the bounds.  (It's the bounds that are being viewpoint-adapted.)  It calls `atTypeDecl` internally, and the mismatch between the method name and what it calls is also surprising.  Can you give it a more evocative name?